client/src/pages/HostGame.tsx

```diff
@@ -30,12 +51,18 @@ const HostGame = () => {
       </div>
 
       <div className="w-screen h-12 mt-8">
-        <button
-          className="text-white bg-primary text-2xl font-bold w-1/3 rounded-xl h-full"
-          onClick={createGame}
-        >
-          Create Game
-        </button>
+        {loading ? (
```

Comment:
Put this ternary on the child inside the same button. If we want to stop the user activating on click event when its loading use the `disabled={loading}` attribute on the button.